Source/services/ClientProcessor.ts

```diff
@@ -0,0 +1,126 @@
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { ConceptAs } from '@dolittle/concepts';
+import { Failure } from '@dolittle/contracts/Protobuf/Failure_pb';
+import { Guid } from '@dolittle/rudiments';
+import { ExecutionContext } from '@dolittle/sdk.execution';
+import { failures } from '@dolittle/sdk.protobuf';
+import { Cancellation, RetryPolicy, retryWithPolicy } from '@dolittle/sdk.resilience';
+import { Observable } from 'rxjs';
+import { repeat } from 'rxjs/operators';
+import { Logger } from 'winston';
+import { IReverseCallClient } from './IReverseCallClient';
+import { RegistrationFailed } from './RegistrationFailed';
+
+
+/**
+ * Defines a system for registering a processor that handles request from the Runtime.
+ */
+export abstract class ClientProcessor<TIdentifier extends ConceptAs<Guid, string>, TRegisterArguments, TRegisterResponse, TRequest, TResponse> {
+    private _pingTimeout = 1;
+
+    /**
+     * Initializes a new {@link ClientProcessor}.
+     * @param {string} _kind What kind of a processor it is.
+     * @param {TIdentifier} _identifier The unique identifier for the processor.
+     * @param {Logger} _logger The logger.
+     */
+    constructor(
+        protected readonly _kind: string,
+        protected readonly _identifier: TIdentifier,
+        protected readonly _logger: Logger) { }
+
+    /**
+     * Registers a processor with the Runtime, and if successful starts handling requests.
+     * @param {Cancellation} cancellation Used to cancel the registration and processing.
+     * @returns {Observable} Representing the connection to the Runtime.
+     */
+    register(cancellation: Cancellation): Observable<void> {
+        const client = this.createClient(
+            this.registerArguments,
+            (request: TRequest, executionContext: ExecutionContext) => this.catchingHandle(request, executionContext),
+            this._pingTimeout,
+            cancellation);
+        return new Observable<void>(subscriber => {
+            this._logger.debug(`Registering ${this._kind} ${this._identifier} with the Runtime.`);
+            client.subscribe({
+                next: (message: TRegisterResponse) => {
+                    const failure = this.getFailureFromRegisterResponse(message);
+                    if (failure) {
+                        subscriber.error(new RegistrationFailed(this._kind, this._identifier.value, failures.toSDK(failure)!));
+                    } else {
+                        this._logger.debug(`${this._kind} ${this._identifier} registered with the Runtime, start handling requests.`);
+                    }
+                },
+                error: (error: Error) => {
+                    subscriber.error(error);
+                },
+                complete: () => {
+                    this._logger.debug(`Registering ${this._kind} ${this._identifier} handling of requests completed.`);
```

Comment:
This is printed when the thing stops? If so I think we can remove the first "Registering" part
```suggestion
                    this._logger.debug(`${this._kind} ${this._identifier} handling of requests completed.`);
```